tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -986,3 +1010,43 @@ func waitForFullSyncToFinish(client clientset.Interface, ctx context.Context,
 	})
 	return waitErr
 }
+
+// checkForEventWithMessage fetches events list of the given object name and checkes for
+// specified message and returns true if expected error message found
+func checkForEventWithMessage(client clientset.Interface, namespace string,
+	name string, expectedMsg string) bool {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	eventFound := false
+	framework.Logf("Checking for error in events related to " + name)
+	eventList, _ := client.CoreV1().Events(namespace).List(ctx,
+		metav1.ListOptions{FieldSelector: fmt.Sprintf("involvedObject.name=%s", name)})
+	for _, item := range eventList.Items {
+		framework.Logf("message: %v", item.Message)
+		if strings.Contains(item.Message, expectedMsg) {
+			framework.Logf("Expected event found. EventList Reason: "+
+				"%q"+" EventList item: %q", item.Reason, item.Message)
+			eventFound = true
+			break
+		}
+	}
+	return eventFound
+}
+
+// checkIfPvDeleteFailWithSpecificMsg checks if a PV delete fails
+func checkIfPvDeleteFailWithSpecificMsg(client clientset.Interface, pv *v1.PersistentVolume) {
+	volumeHandle := pv.Spec.CSI.VolumeHandle
+	err := fpv.WaitForPersistentVolumeDeleted(client, pv.Name, poll,
+		pollTimeout)
+	errMsg := "The object or item referred to could not be found"
+	if err != nil && checkForEventWithMessage(client, "", pv.Name, errMsg) {
+		framework.Logf("Persistent Volume %v still not deleted with err %v", pv.Name, errMsg)
+		// Orphan volumes may be left over here, hence logging those PVs and ignoring the error for now.
+		_ = e2eVSphere.waitForCNSVolumeToBeDeleted(volumeHandle)
+		framework.Logf("Volume %v still not deleted from CNS with err %v", pv.Name, errMsg)
```

Comment:
what are trying to do here?
only in case of error, we should log  `framework.Logf("Volume %v still not deleted from CNS with err %v", pv.Name, errMsg)`, I think

Reply:
I have corrected the log statement to : framework.Logf("Volume %v still not deleted from CNS with err %v, ignoring it", pv.Name, errMsg), as it is still a valid error and we are choosing to ignore it. In case of any other error, we are failing the test.